packages/EasyErrorHandler/src/Processors/ErrorCodesGroupProcessor.php

```diff
@@ -7,24 +7,34 @@
 use EonX\EasyErrorHandler\DataTransferObjects\ErrorCodeCategoryDto;
 use EonX\EasyErrorHandler\DataTransferObjects\ErrorCodesDto;
 use EonX\EasyErrorHandler\Interfaces\ErrorCodesGroupProcessorInterface;
-use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface;
 
 final class ErrorCodesGroupProcessor implements ErrorCodesGroupProcessorInterface
 {
+    /**
+     * @param array<\EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface> $errorCodesProviders
+     */
     public function __construct(
-        private int $categorySize,
-        private ErrorCodesProviderInterface $errorCodesProvider
+        private readonly int $categorySize,
+        private readonly iterable $errorCodesProviders
```

Comment:
I would recommend to convert this iterable to a real array, because technically it will be a Generator and the code will be able to iterate through it only once so any attempt to list error codes with the same application instance will break.

Reply:
Tagged iterator replaced with array